advanced/am-pattern-4/README.md

```diff
@@ -0,0 +1,302 @@
+
+If you are using WSO2 product Docker images available from WSO2 Private Docker Registry,
+please provide your WSO2 Subscription credentials via input values (using `--set` argument). 
+
+Please see the following example.
+
+```
+ helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 3.1.0-3 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
+```
+
+#### Install Chart From Source
+
+>In the context of this document, <br>
+>* `KUBERNETES_HOME` will refer to a local copy of the [`wso2/kubernetes-apim`](https://github.com/wso2/kubernetes-apim/)
+Git repository. <br>
+>* `HELM_HOME` will refer to `<KUBERNETES_HOME>/advanced`. <br>
+
+##### Clone the Helm Resources for WSO2 API Manager Git repository.
+
+```
+git clone https://github.com/wso2/kubernetes-apim.git
+```
+
+##### Deploy Helm chart for WSO2 API Manager Pattern 2 deployment.
+
+ Helm version 2
+
+ ```
+ helm install --dep-up --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 3.1.0-3 --namespace <NAMESPACE>
+ ```
+
+ Helm version 3
+
+ - Create the Kubernetes Namespace to which you desire to deploy the Kubernetes resources.
+ 
+    ```
+    kubectl create ns <NAMESPACE>
+    ```
+
+ - Deploy the Kubernetes resources using the Helm Chart
+ 
+    ```
+    helm install <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 3.1.0-3 --namespace <NAMESPACE> --dependency-update
+    ```
+
+The above steps will deploy the deployment pattern using WSO2 product Docker images available at DockerHub.
+
+If you are using WSO2 product Docker images available from WSO2 Private Docker Registry,
+please provide your WSO2 Subscription credentials via input values (using `--set` argument). 
+
+Please see the following example.
+
+```
+ helm install --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 3.1.0-3 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
+```
+
+### 2. Obtain the external IP
+
+Obtain the external IP (`EXTERNAL-IP`) of the API Manager Ingress resources, by listing down the Kubernetes Ingresses.
+  
+```  
+kubectl get ing -n <NAMESPACE>
+```
+  
+The output under the relevant column stands for the following.
+
+API Manager Publisher-DevPortal
+
+- NAME: Metadata name of the Kubernetes Ingress resource (defaults to `wso2am-pattern-4-am-ingress`)
+- HOSTS: Hostname of the WSO2 API Manager service (`<wso2.deployment.am.pubDevPortalTM.hostname>`)
+- ADDRESS: External IP (`EXTERNAL-IP`) exposing the API Manager service to outside of the Kubernetes environment
+- PORTS: Externally exposed service ports of the API Manager service
+
+API Manager Gateway
+
+- NAME: Metadata name of the Kubernetes Ingress resource (defaults to `wso2am-pattern-4-am-gateway-external-ingress & wso2am-pattern-4-am-gateway-internal-ingress`)
```

Comment:
@chathuraswso2 has to be `Metadata names`.

Reply:
https://github.com/wso2/kubernetes-apim/pull/444/commits/088623390e99bf605cb653d258ee715430d7043d